chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/pendingListing/reviewer/edition2015/RequiredData2015Reviewer.java

```diff
@@ -337,10 +345,18 @@ public void review(final PendingCertifiedProductDTO listing) {
                                     try {
                                         Integer.valueOf(task.getTaskPathDeviationOptimal());
                                     } catch (final Exception e) {
-                                        listing.getErrorMessages().add(
-                                                msgUtil.getMessage("listing.criteria.badTestTaskNumber",
-                                                        task.getUniqueId(), "Task Path Deviation Optimal",
-                                                        task.getTaskPathDeviationOptimal()));
+                                        try {
+                                            int val = Math.round(Float.valueOf(task.getTaskPathDeviationOptimal()));
+                                            listing.getWarningMessages().add(
+                                                    msgUtil.getMessage("listing.criteria.roundedTestTaskNumber",
+                                                            task.getUniqueId(), "Task Path Deviation Optimal",
+                                                            task.getTaskPathDeviationOptimal(), String.valueOf(val)));
+                                        } catch (final Exception ex) {
+                                            listing.getErrorMessages().add(
+                                                    msgUtil.getMessage("listing.criteria.badTestTaskNumber",
+                                                            task.getUniqueId(), "Task Path Deviation Optimal",
+                                                            task.getTaskPathDeviationOptimal()));
+                                        }
                                     }
```

Comment:
Is there anyway to convert this to a function, to reduce the repetitive code?  
```
private void functionName(String origValue, String badErrorMessage, String uniqueId, String description, String roundedErrorMessage)
```
If you don't think so, I'll approve it, since it is correct.